scripts/check_site_lib.lua

```diff
@@ -99,7 +101,21 @@ function need_table(varname, subcheck, required)
    return var
 end
 
+function need_one_of(val, array)
+   for _, v in ipairs(array) do
+      if v == var then
+         return true
+      end
+   end
+   return false
+end
+
 function need_string_array(varname, required)
    return assert(pcall(need_array, varname, function(e) assert_type(e, 'string') end, required),
 		 "site.conf error: expected `" .. varname .. "' to be a string array")
 end
+
+function need_array_of(varname, array, required)
+   return assert(pcall(need_array, varname, need_one_of(e, array),required),
```

Comment:
This call is wrong - you need to wrap the `need_one_of` into an anonymous function like it is done in `need_string_array`